### ISSUE TITLE
How to disable run callback query methods?

### ISSUE THREAD

**Issue by boggddan** [telegram-bot-rb/telegram-bot#81]
Hello, 
How to disable run callback query methods?
A user can execute a command that will match with a name method callback query. 
How can I block the execution of this method if it is sent by the command?

```ruby
class Telegram::WebhookController < Telegram::Bot::UpdatesController
  include Telegram::Bot::UpdatesController::CallbackQueryContext

  def settings(*)
    button = Telegram::Bot::Types::InlineKeyboardButton.new(
      text: "🔄 Refresh  settings",
      callback_data: 'refresh_settings:', url: '')
    markup = Telegram::Bot::Types::InlineKeyboardMarkup.new(inline_keyboard: [[button]])

    respond_with :message, text: 'Settings', reply_markup: markup.to_hash
  end

  def callback_query(data)
    respond_with :message, text: 'callback_query'
  end

  def refresh_settings_callback_query(*)
    respond_with :message, text: 'refresh_settings_callback_query'
  end
end
```

logs:
```
Processing by Telegram::WebhookController#refresh_settings_callback_query
  Update: {"update_id":113528902,"message":{"message_id":710,"from":{"id":536172205,"is_bot":false,"first_name":"Богдан","username":"b360124","language_code":"ru"},"chat":{"id":536172205,"first_name":"Богдан","username":"b360124","type":"private"},"date":1526656622,"text":"/refresh_settings_callback_query","entities":[{"offset":0,"length":32,"type":"bot_command"}]}}
Responded with message
Completed in 68ms (ActiveRecord: 0.0ms)
```

![2018-05-18_18h17_25](https://user-images.githubusercontent.com/24915025/40243051-ec1ed914-5ac7-11e8-9d23-a7e5ea0ac60b.png)

Thanks

**Comment by printercu** [comment]
Hi! There are two options now:
- `return unless payload_type == 'callback_query'`. You can implement it in before action, use `throw :abort` to halt request.
- Override `.action_for_command` (https://github.com/telegram-bot-rb/telegram-bot/blob/master/lib/telegram/bot/updates_controller.rb#L111) and make it add prefix/suffix to such actions. For example `"#{cmd.downcase!}_cmd"` will make it use `some_cmd` method for `/some` message, this way name conflicts should not be possible.

I've used 1st method. I'll try to find out solution to add to gem.

**Comment by boggddan** [comment]
Thanks, should you give me examples first methods, when you used `throw :abort`, please?

**Comment by boggddan** [comment]
What do you say about this implementation of the problem?

```ruby
  before_action :check_callback_query

  def check_callback_query
    throw :abort if payload_type == 'message' && payload['text'].end_with?('_callback_query')
  end
```
Thanks

**Comment by printercu** [comment]
This may not work properly for `/some_callback_query param`.  It's better to use `mayload_type == 'message' && action_name.end_with?('_callback_query')`

**Comment by boggddan** [comment]
Thank you, this is good solution. You add it on the next release Bot?

**Comment by printercu** [comment]
I don't think this is the right solution, because it'll require same actions for every such cases. There is already workaround for commands with name of update fields: https://github.com/telegram-bot-rb/telegram-bot/blob/master/lib/telegram/bot/updates_controller.rb#L113

I see two ways to solve it: 1) explicitly call `command :cmd_name` or `command def cmd_name(*)`,
2) add prefix/suffix to command to get method name: `/cmd_name => _cmd_name or cmd_name!`.

Right now i'm leaning towards `cmd_name!`.
